### PR TITLE
BigTable: reimplement scanAllObjects with server-side filtering

### PR DESCRIPTION
**Pull request by adutra**
This commit changes the BigTable implementation of scanAllObjects to use server-side instead of client-side filtering on object types.

This is implemented using a condition filter. Condition filters are not recommended generally because they are slower, but scanAllObjects is not meant to be particularly efficient. The fact that we are also limiting the query to a row prefix should alleviate the performance impact.

A quick experiment where I exported 1500 repos containing roughly 350K commits total showed a slight improvement in export time (export uses `scanAllObjects` with a filter on commit object types).

Without this fix: 2 minutes 46 seconds
With this fix: 2 minutes 24 seconds

### CODE REVIEW

File: versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTablePersist.java [hunk 1]
```diff
@@ -568,40 +568,50 @@ public CloseableIterator<Obj> scanAllObjects(
...
+                  .filter(FILTERS.qualifier().exactMatch(QUALIFIER_OBJ_TYPE))
+                  .filter(FILTERS.value().exactMatch(OBJ_TYPE_VALUES[type.ordinal()])));
+        } else {
+          all = false;
+        }
+      }
+      if (!all) {
+        // Edge case: the predicate did not match any object type
+        if (typeFilter == null) {
+          this.paginator = null;
```
Comment (snazy) [inline versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTablePersist.java:603]: Isn't this rather an error than an edge case?

**Comment by snazy** [comment]
Were the times collected via the BT emulator or a real BT cluster?

**Comment by adutra** [comment]
> Were the times collected via the BT emulator or a real BT cluster?

Real BT cluster, but with one single node. Do you want me to collect numbers using 3 or more nodes?

**Comment by snazy** [comment]
> Real BT cluster, but with one single node. Do you want me to collect numbers using 3 or more nodes?

Nah - just wanted to know.

File: versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTablePersist.java (hunk 1 above)
Comment (adutra) [reply versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTablePersist.java:603]: It's up for debate. Do you prefer to throw here?

Comment (snazy) [reply versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTablePersist.java:603]: It's either "all types" or an error IMO.

Comment (adutra) [reply versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTablePersist.java]: OK for error then.

Comment (snazy) [reply versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTablePersist.java]: It's rather an internal thingy - but an undefined situation ATM. Let's defer to another PR (I'd be in favor of falling back to "all types" actually).

**Pull request merged**